roles/docker/tasks/main.yml

```diff
@@ -12,7 +12,7 @@
   register: start_result
 
 - set_fact:
-    docker_service_status_changed = start_result | changed
+    docker_service_status_changed: start_result | changed
```

Comment:
@abutcher, taking the 1st example from #1095:

```
- set_fact:
    etcd_service_status_changed = "{{ start_result | changed }}"
```
(Note the `=` you used in #1095, vs `:` I've used.)

is in fact parsed by YAML as:

```
- 
  set_fact: "etcd_service_status_changed = \"{{ start_result | changed }}\""
```

Whereas:

```
- set_fact:
    docker_service_status_changed: start_result | changed
```

is parsed as 

```
- 
  set_fact: 
    docker_service_status_changed: "start_result | changed"
```

So I don't think it needs to be templated.  And according to my tests, it is firing.  The case that does not fire is:

```
- set_fact:
    etcd_service_status_changed = start_result | changed
```